File: 03. Reversed Addition/Solution.java
/**
  * STATS:
  *     Runtime - 0.44
  *     Memory - 4468 MB
  *     Language - Java
  */
  
import java.util.*;
import java.lang.*;

class Main {
	public static void main (String[] args) throws java.lang.Exception	{
		Scanner scanner = new Scanner(System.in);
		
		//The first integer will signify the number of test cases...
		int testCases = scanner.nextInt();
		
		while (testCases-- > 0)	{
			//The next two space-separated integers are the integers which are to be reversed and then added and then reversed again.
			System.out.println(reverse(reverse(scanner.nextInt()) + reverse(scanner.nextInt())));
		}
	}
	
	public static long reverse(long in)	{
		/**
		  * This method will take a number as the input and return the reverse of that number as the output.
		  */
		
		// Creating a variable to store the result and inititalzing the variable.
		long answer = 0;
		
		while (in > 0)	{
			answer *= 10;
			
			answer += (in % 10);
			in /= 10;
		}
		
		// By now `answer` will contain the reverse of `in`. Returning the same.
		return answer;
	}
}

File: 03. Reversed Addition/Solution.cs
/**
  * STATS:
  * 	Runtime - 0.07
  *		Memory - 18 MB
  *		Langauge - C#
  */

using System;

public class Test	{
	
	///	<summary>
	///		This method will be used to reverse a number and return the reversed number back to the calling method.
	/// </summary>
	/// <param name="input">The integer that is to be reversed and returned</param>		
	public static int Reverse(int input)	{
		// Creating a new variable to store the reverse of `input`.
		int result = 0;
	
		while (input > 0)	{
			result *= 10;
			
			// Taking the last digit from `input` and adding it to `result`
			result += input % 10;
			
			input/= 10;
		}
		
		// By now, the value inside `result` will be the reverse of `input`. Returning the same.
		return result;
	}
	
	public static void Main()	{
		// Getting the number of testcases from input
		int testCases = Convert.ToInt32(Console.ReadLine());

		while (testCases-- > 0)	{
			// Getting the next two integers from the console. Taking an entire line as the input
			//  and splitting the line based upon space into two arrays and storing these arrays.
			string[] input = Console.ReadLine().Split(' ', 2);
			
			// Now, taking the two arrays and converting them into integers.
			int input01 = Convert.ToInt32(input[0]);
			int input02 = Convert.ToInt32(input[1]);
			
			// Getting the reverse of these two numbers and overwriting their original values.
			input01 = Reverse(input01);
			input02 = Reverse(input02);
			
			// Adding the reversed numbers, reversing their sum and printing the same.
			Console.WriteLine(Reverse(input01 + input02));
		}
	}
}